### ISSUE TITLE
`autoBalance = 1` does not balance

### ISSUE THREAD

**Issue by ezheidtmann** [GeospatialPython/pyshp#48]
I don't have access to ArcMap, but a customer told me that the shapefile I made with `pyshp` had a mismatched count of shapes and records. If `len(w._shapes)` and `len(w.records)` are useful metrics before writing, I think I found a bug:

```python
import shapefile
w = shapefile.Writer()
w.shapetype = shapefile.POLYLINE
w.autoBalance = 1
w.field('myfield', 'C', 10, 0)
w.poly(shapeType=shapefile.POLYLINE, parts=[[[1,2], [2,3]]])
w.record(**{})
assert len(w._shapes) == len(w.records) # fails for me
```

**Comment by micahcochran** [comment]
`autoBalance` is a property of the [Editor class](https://github.com/GeospatialPython/pyshp/blob/master/shapefile.py#L1072).  The Writer class does not have this property.  So, I don't think this could be classified as a bug.

However, it would probably be fairly easy to move from the Editor class into the Writer class.

**Comment by ezheidtmann** [comment]
Perhaps it's a documentation bug then. The README introduces the Writer class like this:

```
w = shapefile.Writer(shapeType=1)
```

Then stresses the importance of balancing with this example:

```
w.autoBalance = 1
```

**Comment by ezheidtmann** [comment]
And maybe it's a problem that `w.record(**{})` does not create a record?

**Comment by micahcochran** [comment]
> Perhaps it's a documentation bug then. 

The documentation doesn't match the implementation.  You hit the nail on the head.

**Comment by ezheidtmann** [comment]
I made a PR for the docs #51 and for the behavior of `w.record()` #50. 

And thanks for the library -- very helpful to avoid installing gdal everywhere.

**Comment by karimbahgat** [comment]
#50 merged and #51 fixed in 6906953acd2caf2d4d6890160344e9d2baea058b
